source/jormungandr/jormungandr/street_network/with_parking.py

```diff
@@ -0,0 +1,128 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+import logging
+from jormungandr.street_network.street_network import AbstractStreetNetworkService
+from jormungandr import street_network, utils
+
+from jormungandr.street_network.asgard import Asgard
+from navitiacommon import response_pb2
+
+
+class WithParking(AbstractStreetNetworkService):
+    def __init__(self, instance, service_url, modes=None, id=None, timeout=10, api_key=None, **kwargs):
+        self.instance = instance
+        self.modes = modes or []
+        self.sn_system_id = id or 'with_parking'
+        self.parking_module = utils.create_object(kwargs.get('parking', None))
+        config = kwargs.get('street_network', None)
+        if 'service_url' not in config['args']:
+            config['args'].update({'service_url': None})
+        if 'instance' not in config['args']:
+            config['args'].update({'instance': instance})
+
+        config['args'].update({'modes': self.modes})
+        self.street_network = utils.create_object(config)
+
+    def status(self):
+        return {'id': unicode(self.sn_system_id), 'class': self.__class__.__name__, 'modes': self.modes}
+
+    def _direct_path(
+        self, mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+    ):
+        response = self.street_network._direct_path(
+            mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type
+        )
+        if response and len(response.journeys):
+            self.add_parking_section_in_direct_path(response, pt_object_destination)
+
+        return response
+
+    def add_parking_section_in_direct_path(self, response, pt_object_destination):
+        logger = logging.getLogger(__name__)
+        logger.info("Creating parking section for direct path")
+
+        for journey in response.journeys:
+            section = journey.sections.add()
+            journey.nb_sections += 1
+
+            # The origin and destination of the parking section is the destination of the journey
+            section.origin.CopyFrom(pt_object_destination)
+            section.destination.CopyFrom(pt_object_destination)
+            # And we have to complete the destination of the first section ourself
+            # Because Jormun does not do it afterwards
+            journey.sections[0].destination.CopyFrom(pt_object_destination)
+
+            section.duration = self.parking_module.park_duration
```

Comment:
the parking module must have a method taking the coords and the datetime that returns the park (resp leave) duration.